### PR TITLE
Rework useSnapshotProposal and usages

### PR DESCRIPTION
**Pull request by DarksightKellar**
While working on typechain fixes, I found some naming conventions hard to follow around proposal actions. While digging through the code, I noticed an old TODO recommending a refactor. Got lost in the sauce, did some light moving around and type tightening, and here we are -- a new PR.

Untested, but pls review. If changes make sense, I plan to test in the typechain branch.

### CODE REVIEW

File: src/types/fractal.ts [hunk 1]
```diff
@@ -295,6 +294,7 @@ export interface NodeHierarchy {
 }
 
 export type FractalProposal = AzoriusProposal | MultisigProposal | SnapshotProposal;
+export type Activity = FractalProposal;
```
[the remark this answers is not in the document]
Comment (DarksightKellar) [reply src/types/fractal.ts:297]: I'll gladly do it in this PR heh

File: src/components/Proposals/SnapshotProposalDetails/index.tsx [hunk 2]
```diff
@@ -21,36 +21,34 @@ export default function SnapshotProposalDetails({ proposal }: ISnapshotProposalD
...
       </GridItem>
-      <GridItem>
-        <SnapshotProposalSummary proposal={extendedSnapshotProposal} />
-        {user.address && (
-          <VoteContextProvider
-            proposal={proposal}
-            extendedSnapshotProposal={extendedSnapshotProposal}
-          >
-            <AzoriusOrSnapshotProposalAction
+      {extendedSnapshotProposal && (
```
Comment (adamgall) [inline src/components/Proposals/SnapshotProposalDetails/index.tsx:36]: Will this affect UX? Not sure, but leaving a comment here regardless.
